dashboard-ui/src/components/SurveyResults/surveyResults.jsx

```diff
@@ -184,20 +184,22 @@ export function SurveyResults() {
 
             setFilteredData(filteredData);
 
-            let indices = [];
+            let scenarios = {}
             for (const result of filteredData) {
                 if (result.results) {
                     for (const x of Object.keys(result.results)) {
                         if (result.results[x]?.scenarioIndex) {
-                            indices.push(result.results[x].scenarioIndex);
+                            const scenarioIndex = result.results[x].scenarioIndex;
+                            const scenarioName = result.results[x]?.scenarioName || `Scenario ${scenarioIndex}`;
+                            scenarios[scenarioIndex] = scenarioName;
                         }
                     }
                 }
             }
-            indices = Array.from(new Set(indices));
-            indices.sort((a, b) => a - b);
-            setScenarioIndices(indices);
-            if (indices.length > 0) {
+            const sortedIndices = Object.keys(scenarios).map(Number).sort((a, b) => a - b);
+            setScenarioIndices(scenarios);
+
+            if (sortedIndices.length > 0) {
```

Comment:
I'm not certain I understand the point of this section. Before, it looks like this was making sure the scenarios would appear in order. Now, it looks like you're sorting the indices just to see if there's at least one key. If you are just looking to see if there's at least one entry in your scenarios object, you should probably just check if Object.keys(scenarios).length > 0. The sorting doesn't seem to be doing anything here, but I might be missing something

Reply:
Thanks for the catch, this should have been changed (leftover from when I was testing). I will change this as the sort doesn't accomplish anything.